Lib/concurrent/futures/_base.py

```diff
@@ -513,6 +517,8 @@ def set_result(self, result):
         Should only be used by Executor implementations and unit tests.
         """
         with self._condition:
+            if self._state in [CANCELLED, CANCELLED_AND_NOTIFIED, FINISHED]:
```

Comment:
Use tuple instead of a list. 
By this, the tuple is stored in code object.
List is recreated every time on the function execution.

Reply:
I would normally use a tuple, but went with a list because that's the style the rest of the module used. Should I update the other methods in Future to use tuples?